### ISSUE TITLE
Implement module providing query context

### ISSUE THREAD

**Issue by wasamasa** [jorgenschaefer/circe#245]
Rationale:

It's common for IRC users to have queries with people they spoke to previously, possibly in an earlier session.  It would be helpful if your client could provide you the previous context automatically to jog your memory and avoid you to have to look up your logs. See also  #244.

Notes:

- Make the size of the context configurable
- Provide a default (but swappable) function that assumes the default logging has been used (and complains if that's not the case), finds the log and inserts backlog into the query buffer
- Other implementations of said function could grab logs from elsewhere (like ZNC logs on a remote server or something querying an API for backlog)

Prior art:

- [queryresume.pl](https://github.com/irssi/scripts.irssi.org/blob/gh-pages/scripts/queryresume.pl) (irssi)
- [queryman.py](http://weechat.org/scripts/source/queryman.py.html/) (weechat)

**Comment by jorgenschaefer** [comment]
Nice idea.

Any particular reason why this should not work for channel logs just as well?

**Comment by wasamasa** [comment]
Not really, other than channel logs containing more information that cannot be fully restored, such as the nicknames for tracking/coloring.

**Comment by jorgenschaefer** [comment]
Query logs also contain nick changes and such.
